File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public String solution(int[] numbers) {
        StringBuilder answer = new StringBuilder();
        List<String> list = Arrays.stream(numbers).boxed().map(String::valueOf).map(s -> s.repeat(3))
          .sorted(Comparator.reverseOrder())
            .collect(Collectors.toUnmodifiableList());
        
        for(String i : list){
    //      String s = String.valueOf(i);
          answer.append(Integer.valueOf(i.substring(0, i.length() / 3)));
        }
        String r = answer.toString();
        if(r.matches("0+")){
          return "0";
        }
        return r;
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.kt
class Solution {
    fun solution(numbers: IntArray): String {
        val result = numbers.sortedBy { -it.toString().repeat(6).substring(0, 6).toInt() }
            .map{ it.toString()}
            .joinToString ("")
        if(result.toSet().size == 1 && result[0] == '0') {
            return "0"
        }
        return result
    }
}